pymrcpp/pyOperators.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ *  \date Mar 21, 2018
+ *  \author Magnar Bjørgve <magnar.bjorgve@uit.no> \n
+ *          Hylleraas Centre for Quantum Molecular Sciences \n
+ *          UiT - The Arctic University of Norway
+ */
+
+
+#include "pybind11/pybind11.h"
+
+#include "trees/MultiResolutionAnalysis.h"
+#include "operators/ABGVOperator.h"
+#include "operators/PoissonOperator.h"
+#include "operators/HelmholtzOperator.h"
+#include "operators/ConvolutionOperator.h"
+
+using namespace mrcpp;
+namespace py = pybind11;
+
+
+template<int D>
+void pyOperators(py::module &m) {
+
+    std::stringstream DerOperatorName;
+    DerOperatorName << "DerivativeOperator" << D << "D";
+    py::class_<DerivativeOperator<D>> deriv(m, DerOperatorName.str().data());
+    deriv
+        .def(py::init<MultiResolutionAnalysis<D>>());
+
+    std::stringstream ABGVOperatorName;
+    ABGVOperatorName << "ABGVOperator" << D << "D";
+    py::class_<ABGVOperator<D>> (m, ABGVOperatorName.str().data(), deriv)
+        .def(py::init< MultiResolutionAnalysis<D> &, double, double >());
+
+
+    std::stringstream ConvOperatorName;
+    ConvOperatorName << "ConvolutionOperator" << D << "D";
+    py::class_<ConvolutionOperator<D>> convop(m, ConvOperatorName.str().data());
+    convop
+        .def(py::init<MultiResolutionAnalysis<D> &, double>());
+
+    if (D==3){
+        py::class_<PoissonOperator> (m, "PoissonOperator", convop)
+            .def(py::init<MultiResolutionAnalysis<3> &, double >());
+        py::class_<HelmholtzOperator> (m, "HelmholtzOperator", convop)
+            .def(py::init<MultiResolutionAnalysis<3> &, double, double>());
+    }
+
+}
+
+template void pyOperators<1>(py::module &m);
```

Comment:
If the only option you have is really to template the binding functions, then instantiation happens at call site, _i.e._ in `pymrcpp.cpp` and you don't need these three lines. Same for the `pyMethods.cpp` and `pyFundamentalTypes.cpp` files.